File: ExpenseTracker.Api/Common/UserRoles.cs
namespace ExpenseTracker.Api.Common;

/// <summary>
/// Canonical role names stored on <see cref="Models.User.Role"/> and issued as JWT role claims.
/// Values are lowercase so <c>[Authorize(Roles = ...)]</c> matches case-sensitively.
/// </summary>
public static class UserRoles
{
    public const string User = "user";
    public const string Admin = "admin";

    public const int MaxLength = 16;

    public static readonly string[] All = [User, Admin];

    public static string Normalize(string? role)
    {
        var normalized = role?.Trim().ToLowerInvariant();
        return string.IsNullOrEmpty(normalized) ? User : normalized;
    }

    public static bool IsValid(string? role)
    {
        var normalized = Normalize(role);
        return normalized is User or Admin;
    }

    public static bool IsAdmin(string? role) => Normalize(role) == Admin;
}

File: patches/flutter-son/lib/core/auth/user_roles.dart
/// Canonical API role names. Keep in sync with ExpenseTracker.Api `UserRoles`.
abstract final class UserRoles {
  static const user = 'user';
  static const admin = 'admin';

  /// Same mailbox as ExpenseTracker.Api `AuthBootstrapDefaults.DevelopmentAdminEmail`.
  static const bootstrapAdminEmail = 'boss@test.com';

  static String normalize(String? role) {
    final value = role?.trim().toLowerCase();
    return (value == null || value.isEmpty) ? user : value;
  }

  static bool isAdmin(String? role) => normalize(role) == admin;

  static bool isValid(String? role) {
    final value = normalize(role);
    return value == user || value == admin;
  }
}
